src/bundle/DependencyInjection/Configuration/Parser/UserModeParser.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\AdminUi\DependencyInjection\Configuration\Parser;
+
+use Ibexa\AdminUi\UserSetting\UserMode;
+use Ibexa\Bundle\Core\DependencyInjection\Configuration\AbstractParser;
+use Ibexa\Bundle\Core\DependencyInjection\Configuration\SiteAccessAware\ContextualizerInterface;
+use Symfony\Component\Config\Definition\Builder\NodeBuilder;
+
+/**
+ * Configuration parser for user modes.
+ *
+ * @Example configuration:
+ *
+ * ```yaml
+ * ibexa:
+ *   system:
+ *      default: # configuration per siteaccess or siteaccess group
+ *          admin_ui:
+ *              default_user_mode: smart
+ * ```
+ */
+final class UserModeParser extends AbstractParser
```

Comment:
Since the main key is called `admin_ui` I think it would make more sense to call this Config Parser `AdminUiParser`. I'm pretty sure we will have more configuration added in the future.